pkg/agent/controller/l7visibility/l7visibility_controller_test.go

```diff
@@ -0,0 +1,546 @@
+	informerFactory  informers.SharedInformerFactory
+	localPodInformer cache.SharedIndexInformer
+}
+
+func (c *fakeController) startInformers(stopCh chan struct{}) {
+	c.informerFactory.Start(stopCh)
+	c.informerFactory.WaitForCacheSync(stopCh)
+	go c.localPodInformer.Run(stopCh)
+	go c.namespaceInformer.Run(stopCh)
+	cache.WaitForCacheSync(stopCh, c.localPodInformer.HasSynced, c.namespaceInformer.HasSynced)
+}
+
+func newFakeControllerAndWatcher(t *testing.T, objects []runtime.Object, interfaces []*interfacestore.InterfaceConfig) *fakeController {
+	controller := gomock.NewController(t)
+	mockOFClient := openflowtest.NewMockClient(controller)
+
+	client := fake.NewSimpleClientset(objects...)
+	informerFactory := informers.NewSharedInformerFactory(client, 0)
+	nsInformer := informerFactory.Core().V1().Namespaces()
+
+	localPodInformer := coreinformers.NewFilteredPodInformer(
+		client,
+		metav1.NamespaceAll,
+		0,
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+		func(options *metav1.ListOptions) {
+			options.FieldSelector = fields.OneTermEqualSelector("spec.nodeName", "fakeNode").String()
+		},
+	)
+
+	ifaceStore := interfacestore.NewInterfaceStore()
+	for _, itf := range interfaces {
+		ifaceStore.AddInterface(itf)
+	}
+
+	l7Reconciler := l7engine.NewReconciler()
+	l7w := NewL7VisibilityController(mockOFClient, ifaceStore, localPodInformer, nsInformer, l7Reconciler)
+
+	return &fakeController{
+		L7VisibilityController: l7w,
+		mockOFClient:           mockOFClient,
+		client:                 client,
+		informerFactory:        informerFactory,
+		localPodInformer:       localPodInformer,
+	}
+}
+
+func newPodObject(name, namespace string, annotations map[string]string) *v1.Pod {
+	return &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        name,
+			Namespace:   namespace,
+			Annotations: annotations,
+		},
+		Spec: v1.PodSpec{
+			NodeName: "fakeNode",
+		},
+		Status: v1.PodStatus{
+			PodIP: "10.0.0.1",
+		},
+	}
+}
+
+func newNamespaceObject(name string, annotations map[string]string) *v1.Namespace {
+	return &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        name,
+			Annotations: annotations,
+			Labels: map[string]string{
+				"fakeKey": "fakeValue",
+			},
+		},
+	}
+}
+
+func newPodInterface(podName, podNamespace string, ofPort int32) *interfacestore.InterfaceConfig {
+	containerName := k8s.NamespacedName(podNamespace, podName)
+	return &interfacestore.InterfaceConfig{
+		InterfaceName:            util.GenerateContainerInterfaceName(podName, podNamespace, containerName),
+		ContainerInterfaceConfig: &interfacestore.ContainerInterfaceConfig{PodName: podName, PodNamespace: podNamespace, ContainerID: containerName},
+		OVSPortConfig:            &interfacestore.OVSPortConfig{OFPort: ofPort},
+	}
+}
+
+func waitEvents(t *testing.T, expectedEvents int, c *fakeController) {
+	require.NoError(t, wait.Poll(100*time.Millisecond, 5*time.Second, func() (done bool, err error) {
```

Comment:
I noticed that existing TC unit tests use 10ms and not 100ms. Is there a reason for the difference?

Reply:
nothing as such, changing to 10ms